staging/src/k8s.io/cli-runtime/pkg/genericclioptions/config_flags.go

```diff
@@ -289,12 +289,15 @@ func (f *ConfigFlags) toDiscoveryClient() (discovery.CachedDiscoveryInterface, e
 
 // getDefaultCacheDir returns default caching directory path.
 // it first looks at KUBECACHEDIR env var if it is set, otherwise
+// it'll check if XDG_CACHE_HOME var is set and if so, use that. otherwise
 // it returns standard kube cache dir.
 func getDefaultCacheDir() string {
 	if kcd := os.Getenv("KUBECACHEDIR"); kcd != "" {
 		return kcd
 	}
-
+	if kcd := os.Getenv("XDG_CACHE_HOME"); kcd != "" {
+		return kcd
```

Comment:
`XDG_CACHE_HOME` contains the value of the directory where the different caches are hosted (by default `~/.cache`), but you **have to create a subdirectory** to store your cache. If you return XDG_CACHE_HOME directly then you will not have that subdirectory.

I think that you shoud return:
```go
return filepath.Join(kcd, "kube")
```

Reply:
You're right, not sure how I missed it. Just fixed that.